### PR TITLE
CORE-8025 Renamed 'libs/http-rpc-server' module to 'libs/rest-server'

### PR DESCRIPTION
**Pull request by ben-millar**
### Overview:
Renamed the `libs/http-rpc-server` module to `libs/rest-server` and refactored the contents to replace `rpc` with `rest` in any classes defined within this module. This PR satisfies the [CORE-8025](https://r3-cev.atlassian.net/browse/CORE-8025) task as part of the broader initiative described in [CORE-7040](https://r3-cev.atlassian.net/browse/CORE-7040).

Obsolete function `fun isHttpRpcEnabled(): Boolean` was removed from `RestServerSettingsProvider`; cleared with @vkolomeyko .

### Testing:
Local tests are passing both as part of the composite build (`./gradlew publishOSGiImage`), and as a standalone build (`./gradlew build`)

[CORE-8025]: https://r3-cev.atlassian.net/browse/CORE-8025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-7040]: https://r3-cev.atlassian.net/browse/CORE-7040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/http-rpc/rest-server/src/main/kotlin/net/corda/httprpc/server/config/RestServerSettingsProvider.kt [hunk 1]
```diff
@@ -4,13 +4,9 @@ import net.corda.utilities.NetworkHostAndPort
 import java.nio.file.Path
 
 /**
- * Interface that provides a way to retrieve HTTP RPC config values, agnostic of particular config providing implementation.
+ * Interface that provides a way to retrieve REST server config values, agnostic of particular config providing implementation.
  */
-interface HttpRpcSettingsProvider {
-    /**
-     * @return whether the HTTP RPC functionality is enabled and the server is accessible.
```
Comment (pmck91) [inline libs/http-rpc/rest-server/src/main/kotlin/net/corda/httprpc/server/config/RestServerSettingsProvider.kt:11]: comment still referring to http rpc

**Review by pmck91 — DISMISSED** [review]
LGTM just one comment

File: libs/http-rpc/rest-server/src/main/kotlin/net/corda/httprpc/server/config/RestServerSettingsProvider.kt (hunk 1 above)
Comment (ben-millar) [reply libs/http-rpc/rest-server/src/main/kotlin/net/corda/httprpc/server/config/RestServerSettingsProvider.kt:11]: This is removed in the delta 😁

**Review by mag1c-48 — DISMISSED** [review]
LGTM - build

**Review by pmck91 — APPROVED** [review]

**Review by mag1c-48 — APPROVED** [review]
LGTM - build only

**Pull request merged**